Consul/Client.cs

```diff
@@ -174,8 +174,14 @@ private void ConfigureFromEnvironment(UriBuilder consulAddress)
                 {
                     consulAddress.Host = uri.Host;
                 }
-
-                consulAddress.Port = uri.Port;
+                if (envAddr.Contains($"{uri.Host}:{uri.Port}"))
```

Comment:
👍 I'm not a fan of this, but it does the job and  I don't see how we can do it differently.

Reply:
:)